src/allocators/lib/AllocatorLib.sol

```diff
@@ -183,6 +196,28 @@ library AllocatorLib {
                 }
                 let diffBalance := sub(currentBalance, oldBalance)
 
+                let additionalCommitmentAmount := calldataload(add(additionalCommitmentAmounts.offset, mul(i, 0x20)))
+                if gt(additionalCommitmentAmount, oldBalance) {
+                    /// @dev This is NOT a sufficient check to guarantee the user has enough unallocated tokens available for the additional commitment.
+                    ///      The additional committed amounts MUST be verified by the implementation of the allocator library before or after this function is called.
+                    ///      This check will only be used to check if enough tokens are generally available to cover the additional commitment,
+                    ///      not if those available tokens are actually unallocated.
+
+                    mstore(0x00, 0x9a534c61) // InvalidBalanceForAdditionalCommitments()
+                    mstore(0x20, oldBalance)
+                    mstore(0x40, additionalCommitmentAmount)
+                    revert(0x1c, 0x44)
+                }
+
+                // Store the old balance in the previousBalances array
+                mstore(add(previousBalancesPointer, mul(i, 0x20)), oldBalance)
+
+                // Add the additional commitment amount to the difference in balance. This amount must be verifiably unallocated.
+                diffBalance := add(diffBalance, additionalCommitmentAmount)
```

Comment:
is it possible to overflow this value by providing a massive additionalCommitmentAmount?